src/top/metrics.rs

```diff
@@ -181,27 +181,31 @@ pub fn subscribe(client: SubscriptionClient, tx: state::EventTx, interval: i64)
 pub async fn init_components(client: &Client) -> Result<state::State, ()> {
     // Execute a query to get the latest components, and aggregate metrics for each resource
     let rows = client
-        .components_query()
+        .components_query(20)
```

Comment:
Just to confirm, will `vector top` iterate through the pages?

Reply:
No, it currently hard-stops at 20. There _is_ support in tui to implement a scrollable list, but I'd deem that beyond the scope of this current PR -- it's entering into "infinite scroll" UI territory, and the complexities that come with it.

I chose a number here that I assumed would suit common terminal window sizes. I intend to revisit this as use-cases present that have a larger number of components. I'm making the assumption here that 20 covers the majority, but open to changing that as required.

Reply:
I'm not sure how typical this is, but we do have a real-world configuration that has 54 components: https://raw.githubusercontent.com/timberio/vector-test-harness/master/cases/real_world_1_performance/ansible/config_files/vector.pipelines.toml

My suggestion would be:

* Calculate the page size based on terminal height (i.e. how many rows can fit)
* Let the user page down through through the list with `PgUp` and `PgDown` (or other keystrokes) as `top` and `htop` do

I think 20 is too low to assume that it can be static. If we want to hardcode a value for now, I might do something like MAX_INT to match the current behavior of fetching "all" components.

Reply:
IIRC, the version of tui we're currently using has limited support for calculating component dimensions in a way that would help determine row counts. #5289 covers bumping to the latest, which has a new [table implementation](https://github.com/fdehau/tui-rs/issues/418#issue-743360814). The primary motivation there is stylistic, but this may facilitate more fluid layouts/wrapping consideration, which may enable us to calculate an accurate row count.

The existing tui version does already have scroll support (the kitchen sink demo uses it), which I think would make a good follow-up issue to implement. This isn't a new concern to this PR, of course-- the existing implementation of `top` doesn't support scrolling, so the visible component list would be that which fits into the window.

To mimic existing behaviour, I'll update to the max int per your suggestion, and file an issue for the scroll implementation 👍

Reply:
https://github.com/timberio/vector/issues/5883

Reply:
Sounds good, thanks!